crates/sui-core/src/authority.rs

```diff
@@ -4374,7 +4374,95 @@ impl AuthorityState {
             })
     }
 
-    fn choose_protocol_version_and_system_packages(
+    fn is_protocol_version_supported_v2(
```

Comment:
I understand v1 and v2 uses different internal data structures, but do you think it's possible to find a way to dedup some code?

Reply:
I did originally have a significantly dedupped version that was based on translated v1 to v2 when loading it from the database. But:

1. It was hard to feel confident that that code behaved 100% identically, and therefore couldn't cause a fork
2. v1 can be deleted once v2 is rolled out